app/src/main/java/org/fossasia/openevent/general/speakers/SpeakerFragment.kt

```diff
@@ -0,0 +1,153 @@
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        rootView = layoutInflater.inflate(R.layout.fragment_speaker, container, false)
+
+        setToolbar(activity)
+        setHasOptionsMenu(true)
+
+        speakerViewModel.error
+            .nonNull()
+            .observe(viewLifecycleOwner, Observer {
+                rootView.snackbar(it)
+            })
+
+        speakerViewModel.speaker
+            .nonNull()
+            .observe(viewLifecycleOwner, Observer {
+                initializeViews(it)
+            })
+
+        speakerViewModel.loadSpeaker(safeArgs.speakerId)
+
+        return rootView
+    }
+
+    private fun initializeViews(speaker: Speaker) {
+        when (speaker.name.isNullOrBlank()) {
+            true -> rootView.name.isVisible = false
+            false -> {
+                rootView.name.text = speaker.name
+                setToolbar(activity, speaker.name)
+            }
+        }
+
+        when (speaker.position.isNullOrBlank()) {
+            true -> rootView.position.isVisible = false
+            false -> rootView.position.text = speaker.position
+        }
+
+        when (speaker.organisation.isNullOrBlank()) {
+            true -> rootView.organization.isVisible = false
+            false -> rootView.organization.text = speaker.organisation
+        }
+
+        if (speaker.longBiography.isNullOrBlank()) {
+            if (speaker.shortBiography.isNullOrBlank())
+                rootView.aboutSpeakerContainer.isVisible = false
+            else rootView.bio.text = speaker.shortBiography.stripHtml()
+        } else rootView.bio.text = speaker.longBiography.stripHtml()
+
+        if (speaker.email.isNullOrBlank() && speaker.mobile.isNullOrBlank() &&
+            speaker.city.isNullOrBlank() && speaker.country.isNullOrBlank() &&
+            speaker.website.isNullOrBlank() && speaker.twitter.isNullOrBlank() &&
+            speaker.facebook.isNullOrBlank() && speaker.linkedin.isNullOrBlank() &&
+            speaker.github.isNullOrBlank()) rootView.speakerContactsContainer.isVisible = false
+        else {
+            when (speaker.email.isNullOrBlank()) {
+                true -> rootView.email.isVisible = false
+                false -> rootView.email.text = "Email: " + speaker.email
+            }
+
+            when (speaker.mobile.isNullOrBlank()) {
+                true -> rootView.mobile.isVisible = false
+                false -> rootView.mobile.text = "Phone Number: " + speaker.mobile
+            }
+
+            if (!speaker.city.isNullOrBlank() && !speaker.country.isNullOrBlank()) {
+                rootView.from.text = "From: " + speaker.city + ", " + speaker.country
+            } else if (speaker.city.isNullOrBlank() && !speaker.country.isNullOrBlank()) {
+                rootView.from.text = "From: " + speaker.country
+            } else if (!speaker.city.isNullOrBlank() && speaker.country.isNullOrBlank()) {
```

Comment:
No need for `speaker.country.isNullOrBlank()` . If city is not null or blank . It will come to this case only.  Extra check . 

Similarly for `speaker.citu.isNullOrBlank` in next else If